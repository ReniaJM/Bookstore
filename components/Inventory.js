import React, {Component}  from 'react';
import BookView from "./BookView";
import {fbase} from "../fbase";

class Inventory extends Component {

    constructor(){
        super();
        this.state = {
            books:[]
        }
    }

    componentDidMount() {
        this.ref= fbase.syncState('bookstore/books', {
            context: this,
            state:'books'
        });
    }

    componentWillUnmount() {
        fbase.removeBinding(this.ref)

    }
    render (){
        const booklisting= this.state.books.map(book => {
             return <BookView  key={book.name} book={book} addOrder={this.props.addOrder}/>
        })

        return (
            <div className='inventory col-md-6'>
                <h2>Lista książek</h2>
                {booklisting}
            </div>
        )
    }
}

export default Inventory