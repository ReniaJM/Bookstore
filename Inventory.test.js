import React from 'react';
import ReactDOM from 'react-dom';
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Inventory from "./components/Inventory";

configure({adapter: new Adapter()})
describe('inventory test', () =>{
    it('inventory renders without crashing', () => {
        const div = document.createElement('div');
        const books = [];
        ReactDOM.render(<Inventory books={books}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('snapshot matches', () =>{
        const books = [];
        const wrapper =shallow (<Inventory books={books}/>)
        expect(wrapper).toMatchSnapshot();
    })
})