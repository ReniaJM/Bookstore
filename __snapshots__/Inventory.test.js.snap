// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inventory test snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Inventory />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        <h2>
          inventory
        </h2>,
      ],
      "className": "inventory col-md-4",
    },
    "ref": null,
    "rendered": Array [
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "inventory",
        },
        "ref": null,
        "rendered": "inventory",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          <h2>
            inventory
          </h2>,
        ],
        "className": "inventory col-md-4",
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "inventory",
          },
          "ref": null,
          "rendered": "inventory",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
